src/client/csfle/client_encryption.rs

```diff
@@ -639,3 +665,70 @@ impl From<String> for EncryptKey {
         Self::AltName(s)
     }
 }
+
+/// A pending `ClientEncryption::create_encrypted_collection` action.
+pub struct CreateEncryptedCollectionAction<'a> {
+    client_enc: &'a ClientEncryption,
+    db: Database,
+    name: String,
+    options: CreateCollectionOptions,
+    master_key: MasterKey,
+}
+
+impl<'a> CreateEncryptedCollectionAction<'a> {
+    /// Execute the collection creation.
+    pub async fn run(self) -> (Document, Result<()>) {
+        let ef = match self.options.encrypted_fields.as_ref() {
+            Some(ef) => ef,
+            None => {
+                return (
+                    doc! {},
+                    Err(Error::invalid_argument(
+                        "no encrypted_fields defined for collection",
+                    )),
+                );
+            }
+        };
+        let mut ef_prime = ef.clone();
+        if let Ok(fields) = ef_prime.get_array_mut("fields") {
+            for f in fields {
+                let f_doc = if let Some(d) = f.as_document_mut() {
+                    d
+                } else {
+                    continue;
+                };
+                if f_doc.get("keyId") == Some(&Bson::Null) {
+                    let d = match self
+                        .client_enc
+                        .create_data_key(self.master_key.clone())
+                        .run()
+                        .await
+                    {
+                        Ok(v) => v,
+                        Err(e) => return (ef_prime, Err(e)),
+                    };
+                    f_doc.insert("keyId", d);
+                }
+            }
+        }
+        let mut opts_prime = self.options.clone();
+        opts_prime.encrypted_fields = Some(ef_prime.clone());
+        (
+            ef_prime,
+            self.db.create_collection(&self.name, opts_prime).await,
+        )
+    }
+
+    /// Set the encrypted fields for the colllection.
+    pub fn encrypted_fields(mut self, encrypted_fields: impl Into<Option<Document>>) -> Self {
+        self.options.encrypted_fields = encrypted_fields.into();
+        self
+    }
+
+    /// Set the complete creation options for the collection.  This will overwrite prior calls to
+    /// `encrypted_fields`.
+    pub fn create_collection_options(mut self, options: CreateCollectionOptions) -> Self {
```

Comment:
Should we take the opportunity here to add action methods for each option rather than accept the options struct? I think that's the direction we're going to want to head in for 3.0. anyway. That way we won't have to deal with multiple encrypted fields maps being passed in.

Reply:
I'd prefer not to - I agree for 3.0 something like that is what we'll want, but for now there are a lot of fields in `CreateCollectionOptions` that aren't directly relevant.

Reply:
If that's the case I'd be inclined to add options methods just for the fields that are relevant or to revert to accepting the options struct as an argument to the method. I think users might be confused as to which way of providing the encrypted fields is correct (i.e. through the options struct or via the separate action method).

I'll defer to you on this though; I think we're in a weird in-between situation when it comes to actions vs. options structs in our API right now, so whatever we end up doing can be ironed out when we get to 3.0.

Reply:
Yeah, makes sense.  I went back to just accepting options as a parameter.